chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/DeveloperManagerImpl.java

```diff
@@ -140,50 +139,55 @@ public DeveloperDTO getById(final Long id) throws EntityRetrievalException {
             throw new EntityCreationException(msg);
         }
 
+        //if any of the statuses (new, old, or any other status in the history) 
+        //is Under Certification Ban by ONC make sure there is a reason given
+        for(DeveloperStatusEventDTO statusEvent: updatedDev.getStatusEvents()) {
+            if(statusEvent.getStatus().getStatusName().equals(
+                    DeveloperStatusType.UnderCertificationBanByOnc.toString()) &&
+               StringUtils.isEmpty(statusEvent.getReason())) {
+                throw new MissingReasonException(msgUtil.getMessage("developer.missingReasonForBan",
+                        DeveloperStatusType.UnderCertificationBanByOnc.toString()));
+            }
+        }
+
         // if the before status is not Active and the user is not ROLE_ADMIN
         // then nothing can be changed
         if (!currDevStatus.getStatus().getStatusName().equals(DeveloperStatusType.Active.toString())
                 && !Util.isUserRoleAdmin()) {
-            LOGGER.error("User " + Util.getUsername() + " does not have ROLE_ADMIN and cannot change developer "
-                    + beforeDev.getName() + " because its status is not Active.");
-            throw new EntityCreationException(
-                    "User without ROLE_ADMIN is not authorized to update an inactive developer.");
+            String msg = msgUtil.getMessage("developer.notActiveNotAdminCantChangeStatus", 
+                    Util.getUsername(), beforeDev.getName()); 
+            LOGGER.error(msg);
+            throw new EntityCreationException(msg);
         }
 
         // if the status history has been modified, the user must be role admin
         // except that an acb admin can change to UnderCertificationBanByOnc
         // triggered by listing status update
-        boolean devStatusHistoryUpdated = isStatusHistoryUpdated(beforeDev, developer);
+        boolean devStatusHistoryUpdated = isStatusHistoryUpdated(beforeDev, updatedDev);
         if (devStatusHistoryUpdated
                 && newDevStatus.getStatus().getStatusName()
                         .equals(DeveloperStatusType.UnderCertificationBanByOnc.toString())
                 && !(Util.isUserRoleAdmin() || Util.isUserRoleAcbAdmin())) {
-            LOGGER.error("User " + Util.getUsername() + " does not have ROLE_ADMIN or ROLE_ACB but may "
-                    + "have tried to change status history for the developer " + beforeDev.getName() + " to include "
-                    + DeveloperStatusType.UnderCertificationBanByOnc.toString());
-            throw new EntityCreationException("User cannot change developer status to "
-                    + DeveloperStatusType.UnderCertificationBanByOnc.toString()
-                    + " without ROLE_ADMIN or ROLE_ACB.");
+            String  msg = msgUtil.getMessage("developer.statusChangeNotAllowedWithoutAdminOrAcb", 
+                    DeveloperStatusType.UnderCertificationBanByOnc.toString());
```

Comment:
I took away the ability of an ACB to automatically trigger the Developer's change in status to Under Certification Ban, but I didn't notice that this check hadn't been updated to support that.